AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2188,14 +2228,17 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
     ASEnvironmentStatePropagateDown(layoutSpec, self.environmentTraitCollection);
     
     layoutSpec.isMutable = NO;
-    ASLayout *layout = [layoutSpec measureWithSizeRange:constrainedSize];
+    ASLayout *layout = [layoutSpec calculateLayoutThatFits:constrainedSize parentSize:constrainedSize.max];
     ASDisplayNodeAssertNotNil(layout, @"[ASLayoutSpec measureWithSizeRange:] should never return nil! %@, %@", self, layoutSpec);
       
     // Make sure layoutableObject of the root layout is `self`, so that the flattened layout will be structurally correct.
     BOOL isFinalLayoutable = (layout.layoutableObject != self);
     if (isFinalLayoutable) {
       layout.position = CGPointZero;
-      layout = [ASLayout layoutWithLayoutableObject:self constrainedSizeRange:constrainedSize size:layout.size sublayouts:@[layout]];
+      layout = [ASLayout layoutWithLayoutableObject:self
```

Comment:
There is no decision about that. I could bring it up. Actually this would be a good time to change something like that as we will change the API of ASLayout anyway (`constrainedSizeRange:` to `constrainedSize:`).

I would suggest we go to `layoutWithLayoutable:constrainedSize:etc:`  though or what are you thoughts dropping the `Layoutable` from it, as most of Apple APIs have it for a class initializer: `[NSString stringWithString:...`